resources.ps1

```diff
@@ -0,0 +1,350 @@
+                                    $outputContext.AuditLogs.Add([PSCustomObject]@{
+                                            Message = "Added orgunits: [$($allOrgUnitIdsForPosition)] to position: [$($currentPosition.InceptionPositionCode)]"
+                                            IsError = $false
+                                        })
+                                }
+                            }
+                            else {
+                                if ($targetPosition.state -ne 20) {
+                                    $body = @{
+                                        $state = 20
+                                    }
+
+                                    $splatUpdatePositionsParams = @{
+                                        Uri         = "$($actionContext.Configuration.BaseUrl)/api/v2/hrm/positions/$($targetPosition.id)"
+                                        Method      = 'PUT'
+                                        Headers     = $Headers
+                                        Body        = ($body | ConvertTo-Json)
+                                        ContentType = 'application/json; charset=utf-8'
+                                    }
+                                    $positionsResponse = Invoke-RestMethod @splatUpdatePositionsParams -Verbose:$false # Exception not found
+                                }
+                                else {
+                                    Write-Verbose 'skipping action orgunit already added to position [$($currentPosition.$TitleCode)]'
+                                }
+                            }
+                        }
+                    }
+                }
+                else {
+                    Write-Verbose "[DryRun] Create [$($resource)] Inception resource, will be executed during enforcement" -Verbose
+                }
+            }
+        }
+    }
+    catch {
+        $outputContext.Success = $false
+        $ex = $PSItem
+        if ($($ex.Exception.GetType().FullName -eq 'Microsoft.PowerShell.Commands.HttpResponseException') -or
+            $($ex.Exception.GetType().FullName -eq 'System.Net.WebException')) {
+            $errorObj = Resolve-InceptionError -ErrorObject $ex
+            $auditMessage = "Could not create Inception resource. Error: $($errorObj.FriendlyMessage)"
+            Write-Verbose "Error at Line '$($errorObj.ScriptLineNumber)': $($errorObj.Line). Error: $($errorObj.ErrorDetails)"
+        }
+        else {
+            $auditMessage = "Could not create Inception resource. Error: $($ex.Exception.Message)"
+            Write-Verbose "Error at Line '$($ex.InvocationInfo.ScriptLineNumber)': $($ex.InvocationInfo.Line). Error: $($ex.Exception.Message)"
+        }
+        $outputContext.AuditLogs.Add([PSCustomObject]@{
+                Message = $auditMessage
+                IsError = $true
+            })
+    }    
+}
+catch {
```

Comment:
Can you explain the purpose of the nested try/catch?

Reply:
Think this the result of copy/paste from original script to template. Removed the nested try/catch